agent/consul/usagemetrics/usagemetrics.go

```diff
@@ -77,9 +88,10 @@ type UsageMetricsReporter struct {
 	metricLabels   []metrics.Label
 	stateProvider  StateProvider
 	tickerInterval time.Duration
+	getMembersFunc getMembersFunc
 }
 
-func NewUsageMetricsReporter(cfg *Config) (*UsageMetricsReporter, error) {
+func NewUsageMetricsReporter(cfg *Config, getMembersFunc getMembersFunc) (*UsageMetricsReporter, error) {
```

Comment:
Do you think it would be a little more consistent if the `getMembersFunc` was specified within the `Config` along with having a `WithMemberGetter` method for building the configuration?

Reply:
I considered putting it in the `Config` type but it felt more like functionality and not configuration? Wouldn't argue with changing it though

Reply:
Went ahead and made this change, feels much cleaner. Thanks for the suggestions!

Reply:
You are 100% right that its functionality and not configuration in the traditional sense. However everywhere else in the Consul code we would put both into the config struct. 

Really the only reason there is a Config type at all here is so we don't have to pass so many parameters into the constructor but can build it up. We could call it Parameters or something to indicate its real purpose too.